lib/query-google/src/main/java/io/redlink/smarti/query/google/GoogleSearchQueryBuilder.java

```diff
@@ -0,0 +1,196 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.stereotype.Component;
+import org.springframework.util.MultiValueMap;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import static io.redlink.smarti.intend.IrLatchTemplate.IR_LATCH;
+
+/**
+ * 
+ */
+@Component
+@EnableConfigurationProperties(GoogleSearchConfiguration.class)
+public final class GoogleSearchQueryBuilder extends QueryBuilder<GoogleSearchConfiguration> {
+
+    EnumSet<Token.Type> IGNORED_TOKEN_TYPES = EnumSet.of(Type.Attribute, Type.Other);
+    private final static float MIN_TOKEN_CONF = 0.1f;
+    private final GoogleSearchConfiguration defaultConfig;
+    
+    @Autowired
+    public GoogleSearchQueryBuilder(GoogleSearchConfiguration searchConfig, TemplateRegistry registry) {
+        this(searchConfig, registry, null);
+    }
+    
+    protected GoogleSearchQueryBuilder(GoogleSearchConfiguration searchConfig, TemplateRegistry registry, String nameSuffix) {
+        super(GoogleSearchConfiguration.class, registry);
+        //sub-classes MUST classes some name suffix!
+        if(!getClass().equals(GoogleSearchQueryBuilder.class) && StringUtils.isBlank(nameSuffix)){
+            throw new IllegalArgumentException("the parsed nameSuffix MUST NOT be NULL nor empty");
+        }
+        this.defaultConfig = searchConfig;
+    }
+
+    @Override
+    protected final void doBuildQuery(GoogleSearchConfiguration config, Template template, Conversation conversation, Analysis analysis) {
+        final GoogleSearchQuery query = buildQuery(config, template, conversation, analysis);
+        if (query != null) {
+            template.getQueries().add(query);
+        }
+    }
+
+    protected final GoogleSearchQuery buildQuery(GoogleSearchConfiguration config, Template template, Conversation conversation, Analysis analysis){
+
+        GoogleSearchQuery query = new GoogleSearchQuery(getCreatorName(config), config.getResult(), config.getDefaults());
+
+        List<String> queryTerms = template.getSlots().stream()
+            .filter(s -> validateSlot(s,analysis, MIN_TOKEN_CONF))
+            .filter(s -> acceptSlot(s, analysis))
+            .map(s -> new ImmutablePair<Slot,Token>(s, analysis.getTokens().get(s.getTokenIndex())))
+            .sorted((e1,e2) -> Float.compare(e2.getValue().getConfidence(),e1.getValue().getConfidence())) //sort by confidence
+            .map(e -> {
+                final Slot s = e.getKey();
+                final Token t = e.getValue();
+                final Collection<String> queryParams = new LinkedList<>();
+                switch (s.getRole()) {
+                case IrLatchTemplate.ROLE_LOCATION:
+                    //TODO: add support location filter
```

Comment:
I think Location would be fine to include in Google Searches.

Reply:
I'm not that familiar with `IrLatchTemplate` could you add the location support to this PR? That would be great.